internal/yggdrasil/yggdrasil.go

```diff
@@ -304,22 +307,49 @@ func (h *Handler) PostDataMessageForDevice(ctx context.Context, params yggdrasil
 			return operations.NewPostDataMessageForDeviceInternalServerError()
 		}
 	case "registration":
-		_, err := h.deviceRepository.Read(ctx, deviceID, h.initialNamespace)
-		if err == nil {
-			return operations.NewPostDataMessageForDeviceOK()
-		}
-
-		if !errors.IsNotFound(err) {
-			return operations.NewPostDataMessageForDeviceInternalServerError()
-		}
 		// register new edge device
 		contentJson, _ := json.Marshal(msg.Content)
 		registrationInfo := models.RegistrationInfo{}
-		err = json.Unmarshal(contentJson, &registrationInfo)
+		err := json.Unmarshal(contentJson, &registrationInfo)
 		if err != nil {
 			return operations.NewPostDataMessageForDeviceBadRequest()
 		}
 		logger.V(1).Info("received registration info", "content", registrationInfo)
+
+		res := models.MessageResponse{
+			Directive: msg.Directive,
+			MessageID: msg.MessageID,
+		}
+		content := models.RegistrationResponse{}
+
+		_, err = h.deviceRepository.Read(ctx, deviceID, h.initialNamespace)
+		if err == nil {
```

Comment:
Shouldn't it be `err != nil`? `err == nil` means that the device has already been registered and in the past we ignored such registration request. From now on is that a way to re-issue a certificate?

Reply:
Certificate  expired, and we need to renew it. device worker already started to get 401 status code